agent/eventhandler/handler.go

```diff
@@ -31,11 +37,22 @@ func HandleEngineEvents(taskEngine engine.TaskEngine, client api.ECSClient, even
 					seelog.Error("Unable to handle state change event. The events channel is closed")
 					break
 				}
-				err := eventhandler.AddStateChangeEvent(event, client)
+				err := handleEngineEvent(event, client, taskHandler, attachmentEventHandler)
 				if err != nil {
 					seelog.Errorf("Handler unable to add state change event %v: %v", event, err)
 				}
 			}
 		}
 	}
 }
+
+func handleEngineEvent(event statechange.Event, client api.ECSClient, taskHandler *TaskHandler,
+	attachmentEventHandler *AttachmentEventHandler) error {
+	if event.GetEventType() == statechange.TaskEvent || event.GetEventType() == statechange.ContainerEvent {
```

Comment:
switch-case might be a better alternative here to improve readability.

Reply:
changed to switch-case